lib/lambda_ethereum_consensus/state_transition/epoch_processing.ex

```diff
@@ -46,4 +47,32 @@ defmodule LambdaEthereumConsensus.StateTransition.EpochProcessing do
     new_state = %BeaconState{state | randao_mixes: new_randao_mixes}
     {:ok, new_state}
   end
+
+  @spec process_slashings(BeaconState.t()) :: {:ok, BeaconState.t()}
+  def process_slashings(%BeaconState{validators: validators, slashings: slashings} = state) do
+    epoch = Accessors.get_current_epoch(state)
+    total_balance = Accessors.get_total_active_balance(state)
+
+    proportional_slashing_multiplier = ChainSpec.get("PROPORTIONAL_SLASHING_MULTIPLIER")
+    epochs_per_slashings_vector = ChainSpec.get("EPOCHS_PER_SLASHINGS_VECTOR")
+    effective_balance_increment = ChainSpec.get("EFFECTIVE_BALANCE_INCREMENT")
+
+    slashed_sum = Enum.reduce(slashings, 0, &+/2)
+    adjusted_total_slashing_balance = min(slashed_sum * proportional_slashing_multiplier, total_balance)
+    increment = effective_balance_increment
+
+    validators_indices = Enum.with_index(validators)
+
+    new_state = Enum.reduce(validators_indices, state, fn {validator, index}, acc ->
+      if validator.slashed and (epoch + div(epochs_per_slashings_vector, 2)) == validator.withdrawable_epoch do
+        penalty_numerator = div(validator.effective_balance, increment) * adjusted_total_slashing_balance
+        penalty = div(penalty_numerator, total_balance) * increment
```

Comment:
Maybe it's this?

```suggestion
        penalty = div(penalty_numerator * increment, total_balance)
```

Reply:
this didn't work, it made the exact same calculations as the previous line